### PR TITLE
[ISSUE-3089] make it possible for storage groups to have name with hyphen

### PR DESCRIPTION
**Pull request by xuesyn**
## Description

[# Issue 3089](https://github.com/apache/iotdb/issues/3089)

This PR has made it possible for storage groups to have name with hyphen by changing regular expression in STORAGE_GROUP_MATCHER (server/src/main/java/org/apache/iotdb/db/conf/IoTDBConfig.java). Corresponding tests are modified to check correctness of this change.

<hr>

This PR has:
- [ ] been self-reviewed.
    - [ ] concurrent read
    - [ ] concurrent write
    - [ ] concurrent read and write 
- [ ] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. 
- [ ] added or updated version, __license__, or notice information
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious 
  for an unfamiliar reader.
- [x] added unit tests or modified existing tests to cover new code paths, ensuring the threshold 
  for code coverage.
- [x] added integration tests.
- [ ] been tested in a test IoTDB cluster.

<hr>

### CODE REVIEW

File: server/src/test/java/org/apache/iotdb/db/integration/IoTDBSimpleQueryIT.java [hunk 1]
```diff
@@ -32,11 +32,7 @@
 import org.junit.Before;
 import org.junit.Test;
 
-import java.sql.Connection;
-import java.sql.DriverManager;
-import java.sql.ResultSet;
-import java.sql.SQLException;
-import java.sql.Statement;
+import java.sql.*;
 import java.util.ArrayList;
```
Comment (HTHou) [inline server/src/test/java/org/apache/iotdb/db/integration/IoTDBSimpleQueryIT.java:36]: Hi, using a wild card is not a good choice. Don't change this.

File: server/src/main/java/org/apache/iotdb/db/conf/IoTDBConfig.java [hunk 2]
```diff
@@ -56,7 +56,7 @@
   private static final String ID_MATCHER =
       "([a-zA-Z0-9/\"[ ],:@#$%&{}()*=?!~\\[\\]\\-+\\u2E80-\\u9FFF_]+)";
 
-  private static final String STORAGE_GROUP_MATCHER = "([a-zA-Z0-9_.\\u2E80-\\u9FFF]+)";
+  private static final String STORAGE_GROUP_MATCHER = "([a-zA-Z0-9_.\\-\\u2E80-\\u9FFF]+)";
```
Comment (HTHou) [inline server/src/main/java/org/apache/iotdb/db/conf/IoTDBConfig.java:60]: Update the user guide?

File: server/src/test/java/org/apache/iotdb/db/integration/IoTDBSimpleQueryIT.java (hunk 1 above)
Comment (xuesyn) [reply server/src/test/java/org/apache/iotdb/db/integration/IoTDBSimpleQueryIT.java:36]: okk

File: server/src/main/java/org/apache/iotdb/db/conf/IoTDBConfig.java (hunk 2 above)
Comment (HTHou) [reply server/src/main/java/org/apache/iotdb/db/conf/IoTDBConfig.java:60]: Hi, there is another document need to be update as well...
https://github.com/apache/iotdb/blob/master/docs/UserGuide/Data-Concept/Data-Model-and-Terminology.md

Comment (xuesyn) [reply server/src/main/java/org/apache/iotdb/db/conf/IoTDBConfig.java:60]: Okay, thx for reminding

**Review by HTHou — APPROVED** [review]
LGTM

**Review by qiaojialin — APPROVED** [review]

**Pull request merged**